alma/lib/Utils/ShareOfCheckoutHelper.php

```diff
@@ -0,0 +1,365 @@
+    {
+        return [
+            self::AMOUNT_KEY => 0,
+            self::COUNT_KEY => 0,
+            self::CURRENCY_KEY => $currency,
+        ];
+    }
+
+    /**
+     * Order Ids validated by date
+     *
+     * @return array
+     */
+    public function getOrderIds()
+    {
+        $orderIds = [];
+        $defaultStatesExcluded = [6, 7, 8];
+        $orderIdsByDate = self::getOrdersIdByDate($this->startTime, $this->endTime);
+        foreach ($orderIdsByDate as $orderId) {
+            $currentOrder = new Order($orderId);
+            if (!in_array((int) $currentOrder->current_state, $defaultStatesExcluded)) {
+                $orderIds[] = $currentOrder->id;
+            }
+        }
+
+        return $orderIds;
+    }
+
+    /**
+     * Order by date
+     *
+     * @param string $date_from
+     * @param string $date_to
+     *
+     * @return array
+     */
+    public static function getOrdersIdByDate($date_from, $date_to)
+    {
+        $sql = 'SELECT `id_order`
+                FROM `' . _DB_PREFIX_ . 'orders`
+                WHERE date_add >= \'' . pSQL($date_from) . '\'
+                AND date_add <= \'' . pSQL($date_to) . '\'
+                    ' . Shop::addSqlRestriction();
+        $result = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS($sql);
+
+        $orders = [];
+        foreach ($result as $order) {
+            $orders[] = (int) $order['id_order'];
+        }
+
+        return $orders;
+    }
+
+    /**
+     * @param $startTime
+     *
+     * @return void
+     */
+    public function setShareOfCheckoutFromDate($startTime)
+    {
+        $this->startTime = $startTime . ' 00:00:00';
+        $this->setShareOfCheckoutToDate($startTime);
+    }
+
+    /**
+     * @param $endTime
+     *
+     * @return void
+     */
+    public function setShareOfCheckoutToDate($endTime)
+    {
+        $this->endTime = $endTime . ' 23:59:59';
+    }
+
+    /**
+     * @return string
+     */
+    private function getShareOfCheckoutFromDate()
+    {
+        if (isset($this->startTime)) {
+            return $this->startTime;
+        }
+
+        return date('Y-m-d', strtotime('yesterday')) . ' 00:00:00';
+    }
+
+    /**
+     * @return string
+     */
+    private function getShareOfCheckoutToDate()
+    {
+        if (isset($this->endTime)) {
+            return $this->endTime;
+        }
+
+        return date('Y-m-d', strtotime('yesterday')) . ' 23:59:59';
+    }
+
+    /**
+     * Get Currency ISO Code by ID
+     *
+     * @param string $id
+     *
+     * @return string
+     */
+    private function getIsoCodeById($id)
+    {
+        $currency = new Currency();
+        if (method_exists(get_parent_class($currency), 'getIsoCodeById')) {
+            return $currency->getIsoCodeById($id);
+        }
+
+        return $currency->getCurrency($id)['iso_code'];
+    }
+
+    /**
+     * Payload Share of Checkout
+     *
+     * @return array
+     */
+    public function getPayload()
```

Comment:
you should add unit tests for this method and the 4th used inside at least

Reply:
@syjust-alma the four getPayload test is done ! :)